src/main/java/org/jabref/gui/texparser/WalkFileTreeTask.java

```diff
@@ -0,0 +1,73 @@
+package org.jabref.gui.texparser;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.jabref.gui.util.BackgroundTask;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WalkFileTreeTask extends BackgroundTask<FileNodeViewModel> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WalkFileTreeTask.class);
+    private static final String TEX_EXT = ".tex";
+
+    private final Path directory;
+
+    public WalkFileTreeTask(Path directory) {
+        this.directory = directory;
+    }
+
+    @Override
+    protected FileNodeViewModel call() {
+        return searchDirectory(directory);
+    }
+
+    private FileNodeViewModel searchDirectory(Path directory) {
+        if (directory == null || !Files.exists(directory) || !Files.isDirectory(directory)) {
+            return null;
```

Comment:
it is probably preferable to throw an IOException or InvalidStateEx, instead of returning null.